### ISSUE TITLE
HttpSource.BasicLookupStrategy.auth.basic issue in 5.0

### ISSUE THREAD

**Issue by mightymax** [cantaloupe-project/cantaloupe#492]
After upgrading from 4.1.6 to 5.0.2 (release builds), username & secret used in the HttpSource.BasicLookupStrategy.auth.basic property values do not work in combination with the KakaduNativeProcessor. To me it seems like the Basic Auth credentials are not forwarded to the OkHttp Client in 5.0.

**cantaloupe.properties example:**
```ini
source.static = HttpSource
HttpSource.allow_insecure = true
HttpSource.lookup_strategy = BasicLookupStrategy
HttpSource.BasicLookupStrategy.url_prefix = https://my-image-server-with-basic-auth/images
HttpSource.BasicLookupStrategy.auth.basic.username = Username
HttpSource.BasicLookupStrategy.auth.basic.secret = Password
HttpSource.chunking.enabled = true
HttpSource.chunking.chunk_size = 512K
HttpSource.chunking.cache.enabled = true
HttpSource.chunking.cache.max_size = 5M
```

This works in 4.1.6, but in 5.0.2, an error occurs:
```
15:06:35.207 [qtp320304382-17] INFO  e.i.l.c.r.i.v.ImageResource - Handling GET /iiif/2/e7eca70d-2979-462a-a431-02498786c244/full/512,/0/default.jpg
15:06:35.211 [qtp320304382-17] DEBUG e.i.l.c.r.i.v.ImageResource - Request headers: Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8; Upgrade-Insecure-Requests: 1; User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:88.0) Gecko/20100101 Firefox/88.0; Connection: keep-alive; Host: 192.168.1.110:8080; Accept-Language: nl,en-US;q=0.7,en;q=0.3; Accept-Encoding: gzip, deflate; DNT: 1
15:06:35.216 [qtp320304382-17] DEBUG e.i.l.c.i.MetaIdentifier - [Raw path component: e7eca70d-2979-462a-a431-02498786c244] -> [decoded: e7eca70d-2979-462a-a431-02498786c244] -> [slashes substituted: e7eca70d-2979-462a-a431-02498786c244]
15:06:35.236 [qtp320304382-17] DEBUG e.i.l.c.i.Identifier - [Raw path component: e7eca70d-2979-462a-a431-02498786c244] -> [decoded: e7eca70d-2979-462a-a431-02498786c244] -> [slashes substituted: e7eca70d-2979-462a-a431-02498786c244]
15:06:35.254 [qtp320304382-17] INFO  e.i.l.c.c.InfoCache - Max InfoCache capacity: 26214 (5% max heap / 4096-byte expected average info size)
15:06:35.464 [qtp320304382-17] DEBUG e.i.l.c.c.FilesystemCache - getSourceImageFile(): /var/cache/cantaloupe/source/0d/15/df/0d15dfd126b24392a84384f857fbe97b 
15:06:35.486 [qtp320304382-17] DEBUG e.i.l.c.s.HttpSource - Requesting HEAD https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244 (extra headers: {})
15:06:36.140 [qtp320304382-17] DEBUG e.i.l.c.r.ImageRequestHandler - Format inferred by unknown processor disagrees with the one supplied by HttpSource (unknown) for e7eca70d-2979-462a-a431-02498786c244; trying again
15:06:36.142 [qtp320304382-17] DEBUG e.i.l.c.r.ImageRequestHandler - Format inferred by unknown processor disagrees with the one supplied by HttpSource (unknown) for e7eca70d-2979-462a-a431-02498786c244; trying again
15:06:36.168 [qtp320304382-17] DEBUG e.i.l.c.p.ProcessorFactory - KakaduNativeProcessor selected for format JPEG2000 (ManualSelectionStrategy offered KakaduNativeProcessor)
15:06:36.169 [qtp320304382-17] DEBUG e.i.l.c.s.HttpSource - Resolved e7eca70d-2979-462a-a431-02498786c244 to https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244
15:06:36.171 [qtp320304382-17] DEBUG e.i.l.c.p.ProcessorConnector - Stream -> StreamProcessor connection between HttpSource and KakaduNativeProcessor
15:06:36.172 [qtp320304382-17] DEBUG e.i.l.c.s.HttpSource - newSeekableStream(): using 524288-byte chunks
15:06:36.228 [qtp320304382-17] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Unexpected HTTP response code: 401
15:06:36.253 [qtp320304382-17] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Unexpected HTTP response code: 401
15:06:36.254 [qtp320304382-17] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Kakadu Core Error:
Code-stream must start with an SOC marker!

```

**Workaround:**
HttpSource.lookup_strategy = ScriptLookupStrategy

delegates.rb:
```ruby
class CustomDelegate
  
  attr_accessor :context
  
  def authorize(options = {})
    true
  end
  
  def httpsource_resource_info(options = {})
    if match = context['identifier'].match(/^([a-z0-9]{8}\-[a-z0-9]{4}\-[a-z0-9]{4}\-[a-z0-9]{4}\-[a-z0-9]{12})(?:\.jp2)?$/)
      uid = match[1]
      # Generate Basic Auth with: 
        # echo -ne "Username:Password" | base64 --wrap 0
      return { 
        "uri" => "https://my-image-server-with-basic-auth/images/#{uid}", 
        # This does also does not work!
        # "username" => "Username",
        # "secret" => "Password",
        "headers" => {
          "Authorization" => "Basic VXNlcm5hbWU6UGFzc3dvcmQ="
        }
      }
    else
      return nil
    end
  end
  
  def redactions(options = {})
    []
  end
  
  def overlay(options = {})
  end
  
  def metadata(options = {})
  end
  
  
  def pre_authorize(options = {})
    true
  end
  
  def extra_iiif2_information_response_keys()
    nil
  end
end

```

**Comment by adolski** [comment]
I can't reproduce this using Cantaloupe 5.0.2:

### Install a simple HTTP server and run it

```sh
$ npm install http-server
$ http-server /path/to/my/images --username myUsername --password mySecret
```

### Configure Cantaloupe

```properties
source.static = HttpSource
HttpSource.BasicLookupStrategy.url_prefix = http://localhost:8080/
HttpSource.BasicLookupStrategy.auth.basic.username = myUsername
HttpSource.BasicLookupStrategy.auth.basic.secret = mySecret
```

This works. If I change the username/secret, Cantaloupe returns HTTP 403, as expected. Also, the HttpSource tests related to basic auth are all passing.

Your log output makes me wonder if your server is enforcing HTTP Basic on HEAD requests. In the event of an auth failure, the request is supposed to fail before a processor is selected. You would expect to see something like:

```
11:02:07.490 [qtp888473870-17] DEBUG e.i.l.c.s.HttpSource - Requesting HEAD http://127.0.0.1:8080/image.jpg [extra headers: ...]
11:02:07.492 [qtp888473870-17] INFO  e.i.l.c.r.ErrorResource - Handling GET /iiif/2/image.jpg/full/300,/0/default.jpg
11:02:07.492 [qtp888473870-17] DEBUG e.i.l.c.r.ErrorResource - Request headers: Accept: */*; User-Agent: curl/7.64.1; Host: localhost:8182
11:02:07.492 [qtp888473870-17] DEBUG e.i.l.c.r.ErrorResource - Base URI assembled from X-Forwarded headers: http://localhost:8182
11:02:07.499 [qtp888473870-17] DEBUG e.i.l.c.r.HandlerServlet - Responded to GET /iiif/2/image.jpg/full/300,/0/default.jpg with HTTP 403 in 15 msec
```

**Comment by mightymax** [comment]
It looks to me there is nothing wrong with Cantaloupe doing Basic Auth until it passes the Stream to the KakaduNativeProcessor. Maybe in your simplified demo you could not rreproduce because this processor is not used?

Let me provide my full log from v5.0.2 and 4.1.6, from start to failure (I only posted the parts which I thought are relevant):

**Cantaloupe 4.1.6 Log**
```
me@my-cantaloup-server:/opt/cantaloupe/cantaloupe-4.1.6$ java -Dcantaloupe.config=cantaloupe.properties  -Xmx2g -jar cantaloupe-4.1.6.war 
Loading config file: /opt/cantaloupe/cantaloupe-4.1.6/cantaloupe.properties
2021-06-03 08:56:21.610:INFO::main: Logging initialized @366ms to org.eclipse.jetty.util.log.StdErrLog
2021-06-03 08:56:21.764:INFO:oejs.Server:main: jetty-9.4.24.v20191120; built: 2019-11-20T21:37:49.771Z; git: 363d5f2df3a8a28de40604320230664b9c793c16; jvm 11.0.11+9-post-Debian-1deb10u1
2021-06-03 08:56:24.655:INFO:oejw.StandardDescriptorProcessor:main: NO JSP Support for /, did not find org.eclipse.jetty.jsp.JettyJspServlet
2021-06-03 08:56:25.419:INFO:oejs.session:main: DefaultSessionIdManager workerName=node0
2021-06-03 08:56:25.420:INFO:oejs.session:main: No SessionScavenger set, using defaults
2021-06-03 08:56:25.424:INFO:oejs.session:main: node0 Scavenging every 600000ms
Loading config file: /opt/cantaloupe/cantaloupe-4.1.6/cantaloupe.properties
08:56:25.592 [main] INFO  e.i.l.c.ApplicationContextListener - Debian OpenJDK 64-Bit Server VM 11.0.11 / mixed mode, sharing
08:56:25.600 [main] INFO  e.i.l.c.ApplicationContextListener - Java home: /usr/lib/jvm/java-11-openjdk-amd64
08:56:25.603 [main] INFO  e.i.l.c.ApplicationContextListener - Java library path: /usr/local/lib::/usr/java/packages/lib:/usr/lib/x86_64-linux-gnu/jni:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/jni:/lib:/usr/lib
08:56:25.605 [main] INFO  e.i.l.c.ApplicationContextListener - 2 available processor cores
08:56:25.605 [main] INFO  e.i.l.c.ApplicationContextListener - Heap total: 95MB; max: 2048MB
08:56:25.606 [main] INFO  e.i.l.c.ApplicationContextListener - Effective temp directory: /tmp
08:56:25.608 [main] WARN  e.i.l.c.ApplicationContextListener - The effective temp directory is the system temp directory. Severe errors may result if the application is left running for a long time. Consider setting `temp_pathname` for long-running servers.
08:56:25.612 [main] INFO  e.i.l.c.ApplicationContextListener - 🍈 Starting Cantaloupe 4.1.6
08:56:25.632 [main] INFO  e.i.l.c.s.HeapInvocationCache - Invocation cache limit: 1048576
08:56:26.128 [main] INFO  e.i.l.c.s.DelegateProxy - Loading script code
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules (file:/tmp/jetty-0_0_0_0-8182-cantaloupe-4_1_6_war-_-any-5063771421068425406.dir/webapp/WEB-INF/lib/backport9-1.3.jar) to method sun.nio.ch.NativeThread.signal(long)
WARNING: Please consider reporting this to the maintainers of com.headius.backport9.modules.Modules
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
08:56:33.225 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.BMP: com.sun.imageio.plugins.bmp.BMPImageReader, com.sun.media.imageioimpl.plugins.bmp.BMPImageReader
08:56:33.226 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.DICOM: 
08:56:33.228 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.GIF: com.sun.imageio.plugins.gif.GIFImageReader
08:56:33.232 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.JPEG2000: com.sun.media.imageioimpl.plugins.jpeg2000.J2KImageReader
08:56:33.237 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.JPEG: com.sun.imageio.plugins.jpeg.JPEGImageReader
08:56:33.237 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.PDF: 
08:56:33.240 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.PNG: com.sun.imageio.plugins.png.PNGImageReader
08:56:33.399 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.TIFF: it.geosolutions.imageioimpl.plugins.tiff.TIFFImageReader, com.sun.media.imageioimpl.plugins.tiff.TIFFImageReader, com.sun.imageio.plugins.tiff.TIFFImageReader
08:56:33.400 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers for Format.WebP: 
08:56:33.408 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers for Format.GIF: com.sun.media.imageioimpl.plugins.gif.GIFImageWriter, com.sun.imageio.plugins.gif.GIFImageWriter
08:56:33.413 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers for Format.JPEG2000: com.sun.media.imageioimpl.plugins.jpeg2000.J2KImageWriter
08:56:33.416 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers for Format.JPEG: com.sun.imageio.plugins.jpeg.JPEGImageWriter
08:56:33.416 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers for Format.PDF: 
08:56:33.419 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers for Format.PNG: com.sun.imageio.plugins.png.PNGImageWriter
08:56:33.450 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers for Format.TIFF: it.geosolutions.imageioimpl.plugins.tiff.TIFFImageWriter, com.sun.imageio.plugins.tiff.TIFFImageWriter, com.sun.media.imageioimpl.plugins.tiff.TIFFImageWriter
08:56:33.451 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers for Format.WebP: 
2021-06-03 08:56:33.479:INFO:oejsh.ContextHandler:main: Started o.e.j.w.WebAppContext@5c46a957{Cantaloupe,/,file:///tmp/jetty-0_0_0_0-8182-cantaloupe-4_1_6_war-_-any-5063771421068425406.dir/webapp/,AVAILABLE}{/opt/cantaloupe/cantaloupe-4.1.6/cantaloupe-4.1.6.war}
2021-06-03 08:56:33.503:INFO:oejs.AbstractConnector:main: Started ServerConnector@56f4468b{HTTP/1.1,[http/1.1]}{0.0.0.0:8182}
2021-06-03 08:56:33.504:INFO:oejs.Server:main: Started @12262ms
08:56:50.358 [qtp1873653341-16] DEBUG e.i.l.c.i.Identifier - Raw path component: e7eca70d-2979-462a-a431-02498786c244 -> decoded: e7eca70d-2979-462a-a431-02498786c244 -> scale constraint stripped: e7eca70d-2979-462a-a431-02498786c244 -> slashes substituted: e7eca70d-2979-462a-a431-02498786c244
08:56:50.360 [qtp1873653341-16] INFO  e.i.l.c.r.i.v.ImageResource - Handling GET /iiif/2/e7eca70d-2979-462a-a431-02498786c244/full/!170,170/0/default.jpg
08:56:50.361 [qtp1873653341-16] DEBUG e.i.l.c.r.i.v.ImageResource - Request headers: Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8; Upgrade-Insecure-Requests: 1; User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:88.0) Gecko/20100101 Firefox/88.0; Connection: keep-alive; Host: 192.168.1.110:8182; Accept-Language: nl,en-US;q=0.7,en;q=0.3; Accept-Encoding: gzip, deflate; DNT: 1
08:56:50.377 [qtp1873653341-16] INFO  e.i.l.c.c.InfoCache - Max InfoCache capacity: 1431656 (10% max heap / 150-byte expected average info size)
08:56:50.441 [qtp1873653341-16] DEBUG e.i.l.c.s.DelegateProxy - Instantiated delegate object in 61 msec
08:56:50.453 [qtp1873653341-16] DEBUG e.i.l.c.c.CacheFactory - getSourceCache(): implementation changed; creating a new instance
08:56:50.735 [qtp1873653341-16] DEBUG e.i.l.c.s.HttpSource - Requesting HEAD https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244 (extra headers: (none))
08:56:56.270 [qtp1873653341-16] DEBUG e.i.l.c.p.ProcessorFactory - KakaduNativeProcessor selected for format JP2 (ManualSelectionStrategy offered KakaduNativeProcessor, Java2dProcessor)
08:56:56.273 [qtp1873653341-16] DEBUG e.i.l.c.s.HttpSource - Resolved e7eca70d-2979-462a-a431-02498786c244 to https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244
08:56:56.276 [qtp1873653341-16] DEBUG e.i.l.c.p.ProcessorConnector - StreamSource -> StreamProcessor connection between HttpSource and KakaduNativeProcessor
08:56:56.277 [qtp1873653341-16] DEBUG e.i.l.c.s.HttpSource - newSeekableStream(): using 524288-byte chunks
08:56:56.480 [qtp1873653341-16] DEBUG e.i.l.c.c.InfoService - readInfo(): read e7eca70d-2979-462a-a431-02498786c244 from KakaduNativeProcessor in 193 msec
08:56:56.524 [qtp1873653341-16] DEBUG e.i.l.c.s.DelegateProxy - invokeUncached(): authorize() returned true for args: (none) in 13 msec
08:56:56.528 [qtp1873653341-16] DEBUG e.i.l.c.r.i.v.ImageResource - Base URI assembled from X-Forwarded headers: http://192.168.1.110:8182
08:56:56.531 [qtp1873653341-16] DEBUG e.i.l.c.r.ImageRepresentation - Derivative cache not available; writing directly to the response
08:56:56.541 [qtp1873653341-16] DEBUG e.i.l.c.p.c.JPEG2000KakaduImageReader - Rendered region 0,0/171x130; source 0,0/2720x2069; 4x reduction factor; differential scale 6908814/10000000; PPI 600.0x600.0
08:56:56.583 [qtp1873653341-16] DEBUG e.i.l.c.p.c.JPEGImageWriter - ImageIO plugin preferences: com.sun.imageio.plugins.jpeg.JPEGImageWriter
08:56:56.584 [qtp1873653341-16] DEBUG e.i.l.c.p.c.JPEGImageWriter - Using com.sun.imageio.plugins.jpeg.JPEGImageWriter
08:56:56.642 [qtp1873653341-16] DEBUG e.i.l.c.p.Java2DUtil - removeAlpha(): executed in 54 msec
08:56:56.643 [qtp1873653341-16] DEBUG e.i.l.c.p.c.JPEGImageWriter - Quality: 80; progressive: true
08:56:56.670 [qtp1873653341-16] DEBUG e.i.l.c.r.ImageRepresentation - KakaduNativeProcessor processed in 139 msec: e7eca70d-2979-462a-a431-02498786c244_1:1_scale:!170,170,bicubic_encode:jpg_JPEG_80_interlace_#FFFFFF_8
08:56:56.674 [qtp1873653341-16] DEBUG e.i.l.c.s.s.HTTPImageInputStream - Downloaded 1 chunks (524288 (24.06%) of 2179033 bytes); read 12.72% of chunk data; 0 cache hits; 1 cache misses
08:56:56.674 [qtp1873653341-16] DEBUG e.i.l.c.r.HandlerServlet - Responded to GET /iiif/2/e7eca70d-2979-462a-a431-02498786c244/full/!170,170/0/default.jpg with HTTP 200 in 6377 msec

```
**Cantaloupe 5.0.2 Log**
```
me@my-cantaloup-server:/opt/cantaloupe/cantaloupe-5.0.2$ java -Dcantaloupe.config=./cantaloupe.properties -Xmx2g -jar ./cantaloupe-5.0.2.jar
Loading config file: /opt/cantaloupe/cantaloupe-5.0.2/./cantaloupe.properties
08:42:40.783 [main] INFO  e.i.l.c.ApplicationContextListener - Debian OpenJDK 64-Bit Server VM 11.0.11 / mixed mode, sharing
08:42:40.788 [main] INFO  e.i.l.c.ApplicationContextListener - 2 available processor cores
08:42:40.788 [main] INFO  e.i.l.c.ApplicationContextListener - Heap total: 64MB; max: 2048MB
08:42:40.788 [main] INFO  e.i.l.c.ApplicationContextListener - Java home: /usr/lib/jvm/java-11-openjdk-amd64
08:42:40.789 [main] INFO  e.i.l.c.ApplicationContextListener - Java library path: /usr/local/lib::/usr/java/packages/lib:/usr/lib/x86_64-linux-gnu/jni:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/jni:/lib:/usr/lib
08:42:40.800 [main] INFO  e.i.l.c.ApplicationContextListener - JSR-223 script engines: org.jruby.embed.jsr223.JRubyEngineFactory@2ddc9a9f, jdk.nashorn.api.scripting.NashornScriptEngineFactory@298a5e20
08:42:40.802 [main] INFO  e.i.l.c.ApplicationContextListener - Effective temp directory: ./tmp
08:42:40.803 [main] INFO  e.i.l.c.ApplicationContextListener - 🍈 Starting Cantaloupe 5.0.2
08:42:41.363 [main] DEBUG e.i.l.c.i.FormatRegistry - Read 15 bundled formats: avi, bmp, flv, gif, jp2, jpg, mov, mp4, mpg, pdf, png, tif, webm, webp, xpm
08:42:41.525 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O readers (not in preference order):
	Format.JPEG2000: com.sun.media.imageioimpl.plugins.jpeg2000.J2KImageReader
	Format.PNG: com.sun.imageio.plugins.png.PNGImageReader
	Format.TIFF: com.sun.media.imageioimpl.plugins.tiff.TIFFImageReader, it.geosolutions.imageioimpl.plugins.tiff.TIFFImageReader
	Format.JPEG: com.sun.imageio.plugins.jpeg.JPEGImageReader
	Format.BMP: com.sun.imageio.plugins.bmp.BMPImageReader, com.sun.media.imageioimpl.plugins.bmp.BMPImageReader
	Format.PDF: 
	Format.XPM: 
	Format.WebP: 
	Format.GIF: com.sun.imageio.plugins.gif.GIFImageReader
08:42:41.562 [main] DEBUG e.i.l.c.p.c.IIOProviderContextListener - Image I/O writers (not in preference order):
	Format.JPEG2000: com.sun.media.imageioimpl.plugins.jpeg2000.J2KImageWriter
	Format.PNG: com.sun.imageio.plugins.png.PNGImageWriter
	Format.TIFF: com.sun.media.imageioimpl.plugins.tiff.TIFFImageWriter, com.sun.imageio.plugins.tiff.TIFFImageWriter, it.geosolutions.imageioimpl.plugins.tiff.TIFFImageWriter
	Format.JPEG: com.sun.imageio.plugins.jpeg.JPEGImageWriter
	Format.BMP: com.sun.imageio.plugins.bmp.BMPImageWriter, com.sun.media.imageioimpl.plugins.bmp.BMPImageWriter
	Format.PDF: 
	Format.XPM: 
	Format.WebP: 
	Format.GIF: com.sun.imageio.plugins.gif.GIFImageWriter, com.sun.media.imageioimpl.plugins.gif.GIFImageWriter
08:42:41.571 [main] INFO  o.e.j.s.h.ContextHandler - Started o.e.j.s.ServletContextHandler@5fdcaa40{/,null,AVAILABLE}
08:42:41.637 [main] INFO  o.e.j.s.AbstractConnector - Started ServerConnector@79d8407f{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:8080}
08:42:41.640 [main] INFO  o.e.j.s.Server - Started @2075ms
08:43:25.296 [qtp320304382-15] INFO  e.i.l.c.r.i.v.ImageResource - Handling GET /iiif/2/e7eca70d-2979-462a-a431-02498786c244/full/!170,170/0/default.jpg
08:43:25.303 [qtp320304382-15] DEBUG e.i.l.c.r.i.v.ImageResource - Request headers: Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8; Upgrade-Insecure-Requests: 1; User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:88.0) Gecko/20100101 Firefox/88.0; Connection: keep-alive; Host: 192.168.1.110:8080; Accept-Language: nl,en-US;q=0.7,en;q=0.3; Accept-Encoding: gzip, deflate; DNT: 1
08:43:25.314 [qtp320304382-15] DEBUG e.i.l.c.i.MetaIdentifier - [Raw path component: e7eca70d-2979-462a-a431-02498786c244] -> [decoded: e7eca70d-2979-462a-a431-02498786c244] -> [slashes substituted: e7eca70d-2979-462a-a431-02498786c244]
08:43:25.334 [qtp320304382-15] DEBUG e.i.l.c.i.Identifier - [Raw path component: e7eca70d-2979-462a-a431-02498786c244] -> [decoded: e7eca70d-2979-462a-a431-02498786c244] -> [slashes substituted: e7eca70d-2979-462a-a431-02498786c244]
08:43:25.400 [qtp320304382-15] DEBUG e.i.l.c.c.FilesystemCache - getSourceImageFile(): ./cache/source/0d/15/df/0d15dfd126b24392a84384f857fbe97b 
08:43:25.422 [qtp320304382-15] DEBUG e.i.l.c.s.HttpSource - Requesting HEAD https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244 (extra headers: {})
08:43:27.287 [qtp320304382-15] DEBUG e.i.l.c.r.ImageRequestHandler - Format inferred by unknown processor disagrees with the one supplied by HttpSource (unknown) for e7eca70d-2979-462a-a431-02498786c244; trying again
08:43:27.288 [qtp320304382-15] DEBUG e.i.l.c.r.ImageRequestHandler - Format inferred by unknown processor disagrees with the one supplied by HttpSource (unknown) for e7eca70d-2979-462a-a431-02498786c244; trying again
08:43:27.307 [qtp320304382-15] DEBUG e.i.l.c.p.ProcessorFactory - KakaduNativeProcessor selected for format JPEG2000 (ManualSelectionStrategy offered KakaduNativeProcessor)
08:43:27.309 [qtp320304382-15] DEBUG e.i.l.c.s.HttpSource - Resolved e7eca70d-2979-462a-a431-02498786c244 to https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244
08:43:27.312 [qtp320304382-15] DEBUG e.i.l.c.p.ProcessorConnector - Stream -> StreamProcessor connection between HttpSource and KakaduNativeProcessor
08:43:27.313 [qtp320304382-15] DEBUG e.i.l.c.s.HttpSource - newSeekableStream(): using 524288-byte chunks
08:43:27.494 [qtp320304382-15] INFO  e.i.l.c.c.InfoCache - Max InfoCache capacity: 26214 (5% max heap / 4096-byte expected average info size)
08:43:27.551 [qtp320304382-15] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Unexpected HTTP response code: 401
08:43:27.579 [qtp320304382-15] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Unexpected HTTP response code: 401
08:43:27.580 [qtp320304382-15] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Kakadu Core Error:
Code-stream must start with an SOC marker!

08:43:27.586 [qtp320304382-15] DEBUG e.i.l.c.s.s.HTTPImageInputStream - Downloaded 0 chunks (0 (0.00%) of 2179033 bytes); read NaN% of chunk data; 0 cache hits; 2 cache misses
08:43:27.586 [qtp320304382-15] DEBUG e.i.l.c.r.ImageRequestHandler - Format inferred by KakaduNativeProcessor disagrees with the one supplied by HttpSource (jp2) for e7eca70d-2979-462a-a431-02498786c244; trying again
08:43:27.587 [qtp320304382-15] DEBUG e.i.l.c.s.HttpSource - Requesting GET https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244 (extra headers: {Range=bytes=0-31})
08:43:27.875 [qtp320304382-15] DEBUG e.i.l.c.s.HttpSource - Inferred jp2 format from magic bytes for GET https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244
08:43:27.876 [qtp320304382-15] DEBUG e.i.l.c.p.ProcessorFactory - KakaduNativeProcessor selected for format JPEG2000 (ManualSelectionStrategy offered KakaduNativeProcessor)
08:43:27.879 [qtp320304382-15] DEBUG e.i.l.c.s.HttpSource - Resolved e7eca70d-2979-462a-a431-02498786c244 to https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244
08:43:27.879 [qtp320304382-15] DEBUG e.i.l.c.p.ProcessorConnector - Stream -> StreamProcessor connection between HttpSource and KakaduNativeProcessor
08:43:27.880 [qtp320304382-15] DEBUG e.i.l.c.s.HttpSource - newSeekableStream(): using 524288-byte chunks
08:43:27.909 [qtp320304382-15] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Unexpected HTTP response code: 401
08:43:27.939 [qtp320304382-15] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Unexpected HTTP response code: 401
08:43:27.939 [qtp320304382-15] ERROR e.i.l.c.p.c.j.JPEG2000KakaduImageReader - Kakadu Core Error:
Code-stream must start with an SOC marker!
Kakadu Core Error:
Code-stream must start with an SOC marker!

08:43:27.940 [qtp320304382-15] DEBUG e.i.l.c.s.s.HTTPImageInputStream - Downloaded 0 chunks (0 (0.00%) of 2179033 bytes); read NaN% of chunk data; 0 cache hits; 2 cache misses
08:43:27.941 [qtp320304382-15] DEBUG e.i.l.c.r.ImageRequestHandler - Format inferred by KakaduNativeProcessor disagrees with the one supplied by HttpSource (jp2) for e7eca70d-2979-462a-a431-02498786c244; trying again
08:43:27.942 [qtp320304382-15] INFO  e.i.l.c.r.ErrorResource - Handling GET /iiif/2/e7eca70d-2979-462a-a431-02498786c244/full/!170,170/0/default.jpg
08:43:27.944 [qtp320304382-15] DEBUG e.i.l.c.r.ErrorResource - Request headers: Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8; Upgrade-Insecure-Requests: 1; User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:88.0) Gecko/20100101 Firefox/88.0; Connection: keep-alive; Host: 192.168.1.110:8080; Accept-Language: nl,en-US;q=0.7,en;q=0.3; Accept-Encoding: gzip, deflate; DNT: 1
08:43:27.947 [qtp320304382-15] DEBUG e.i.l.c.r.ErrorResource - Base URI assembled from X-Forwarded headers: http://192.168.1.110:8080
08:43:28.041 [qtp320304382-15] DEBUG e.i.l.c.r.HandlerServlet - Responded to GET /iiif/2/e7eca70d-2979-462a-a431-02498786c244/full/!170,170/0/default.jpg with HTTP 501 in 2795 msec
```
**Curl output HEAD request:**
```
me@my-server:~$ curl -I -u Username:Secret https://my-image-server-with-basic-auth/images/e7eca70d-2979-462a-a431-02498786c244
HTTP/1.1 200 
Server: nginx/1.17.6
Date: Thu, 03 Jun 2021 06:52:19 GMT
Content-Type: image/jp2
Content-Length: 2179033
Connection: keep-alive
Set-Cookie: my_server_id=a68979598c96286b0dc219da7df58a53; expires=Thu, 03-Jun-21 07:52:19 GMT; max-age=3600; path=/images
X-UA-Compatible: IE=10; IE=11
Cache-Control: no-cache
X-Content-Type-Options: nosniff
Content-Security-Policy: img-src data: blob: *; default-src blob: *; script-src 'unsafe-inline' 'unsafe-eval' data: *; style-src 'unsafe-inline' *; font-src data: *
X-XSS-Protection: 1; mode=block
ETag: "dec42d645da918665bfdd5e8e2acb3c0"
Content-Disposition: attachment; filename*=UTF-8''filename.jp2
Accept-Ranges: bytes
```

**Comment by mightymax** [comment]
By the way: as you can see in the log files between 4.x and 5.x there seems to be some extra steps for the processor before it actually knows this is a JP2 source. But maybe this is a totally unrelated issue.

**Comment by adolski** [comment]
You are right, I wasn't using the exact same setup.

This is a bug that happens whenever:

1. auth and chunking are both enabled for HttpSource
2. the server requires auth
3. an image format is requested that supports chunking (JP2 or TIFF)
4. a processor is used that supports random access reading (such as KNP)

Although HttpSource honors the auth info, the HTTP client backing the range-supporting input streams that it creates (`OkHttpHTTPImageInputStreamClient`) does not. It does, however, honor any extra headers returned from the `httpsource_resource_info()` delegate method, which is why your workaround works.

Thanks for reporting this. It is fixed now on the `release/5.0` branch for inclusion in 5.0.3.
